base/src/types.rs

```diff
@@ -553,6 +554,16 @@ impl<Id, T> Type<Id, T>
 impl<Id, T> Type<Id, T>
     where T: Deref<Target = Type<Id, T>>
 {
+    pub fn as_function(&self) -> Option<(&T, &T)> {
+        if let Type::App(ref app, ref args) = *self {
+            if let Type::Builtin(BuiltinType::Function) = **app {
+                assert!(args.len() == 2, "ICE: expected a function application with two arguments, found {}", args.len());
```

Comment:
Yep that seems to be correct behavior as it can only handle fully applied functions/functions that have both argument and return value.